src/build.cc

```diff
@@ -900,17 +905,35 @@ bool Builder::ExtractDeps(CommandRunner::Result* result,
     if (content.empty())
       return true;
 
-    DepfileParser deps(config_.depfile_parser_options);
-    if (!deps.Parse(&content, err))
-      return false;
+    if (deps_type == "gcc") {
+      DepfileParser deps(config_.depfile_parser_options);
+      if (!deps.Parse(&content, err))
+        return false;
 
-    // XXX check depfile matches expected output.
-    deps_nodes->reserve(deps.ins_.size());
-    for (vector<StringPiece>::iterator i = deps.ins_.begin();
-         i != deps.ins_.end(); ++i) {
-      uint64_t slash_bits;
-      CanonicalizePath(const_cast<char*>(i->str_), &i->len_, &slash_bits);
-      deps_nodes->push_back(state_->GetNode(*i, slash_bits));
+      // XXX check depfile matches expected output.
+      deps_nodes->reserve(deps.ins_.size());
+      for (vector<StringPiece>::iterator i = deps.ins_.begin();
+           i != deps.ins_.end(); ++i) {
+        uint64_t slash_bits;
+        CanonicalizePath(const_cast<char*>(i->str_), &i->len_, &slash_bits);
+        deps_nodes->push_back(state_->GetNode(*i, slash_bits));
+      }
+    } else if (deps_type == "msvc_source_dependencies") {
+#if !NINJA_ENABLE_CL_SOURCE_DEPENDENCIES
+      *err = string("msvc_source_dependencies is only supported on Windows");
+      return false;
+#else
+      std::set<std::string> includes;
+      if (!ParseCLSourceDependencies(content, err, includes))
+        return false;
+        
+      for (std::set<std::string>::iterator i = includes.begin();
+           i != includes.end(); ++i) {
+        // Like above, ~0 is assuming that with MSVC-parsed headers, it's ok to
+        // always make all backslashes.
+        deps_nodes->push_back(state_->GetNode(*i, ~0u));
```

Comment:
Testing in [CMake MR 7827](https://gitlab.kitware.com/cmake/cmake/-/merge_requests/7827) found that this does not reconcile the lower-case paths reported by `/sourceDependencies` with mixed-case paths in the build manifest.  We also need to `CanonicalizePath` to handle slash differences.

Reply:
I've addressed both of these over in #2600.